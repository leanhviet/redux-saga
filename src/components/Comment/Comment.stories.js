// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import Comment from './index'

const mockData = {
  id: 1,
  name: 'Le Dinh Nam',
  src: '',
  content:
    'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Minus quibusdam omnis iusto laboriosam. Nulla aliquid sit, ipsa blanditiis, minus labore asperiores quia tempore culpa dignissimos necessitatibus quod distinctio facilis magnam!',
  likeNumber: '150',
  dislikeNumber: '300'
}

storiesOf('Comment', module)
  .add('Default', () => <Comment />)
  .add('Contain content', () => <Comment comment={mockData} />)
