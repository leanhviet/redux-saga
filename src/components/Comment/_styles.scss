.comment {
  padding-top: 30px;
  padding-bottom: 30px;
  border-bottom: 1px solid $iron-approx;

  &__menu {
    &__icon {
      text-transform: uppercase;
      color: $gray;
      font-size: 14px;
      margin-right: 40px;
      text-decoration: none;
      cursor: pointer;

      & i {
        margin-right: 10px;
        padding-left: 10px;
      }
    }
  }
}
