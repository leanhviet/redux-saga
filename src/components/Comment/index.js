// @flow
import React from 'react'

// Components
import Profile from '../Profile'

type Props = {
  comment: Object
}

const Comment = (props: Props) => {
  const { src, name, content, likeNumber, dislikeNumber } = props.comment

  return (
    <div className="comment">
      <Profile src={src} name={name} subName={content} />
      <div className="text-right comment__menu">
        <span className="comment__menu__icon">
          {likeNumber}
          <i className="fa fa-thumbs-up" />
        </span>
        <span className="comment__menu__icon">
          <i className="fa fa-thumbs-down" />
          {dislikeNumber}
        </span>
        <span className="comment__menu__icon">reply</span>
      </div>
    </div>
  )
}

Comment.defaultProps = {
  comment: {
    src: '',
    name: 'Le Anh Viet',
    content: '',
    likeNumber: '100',
    dislikeNumber: '50'
  }
}

export default Comment
