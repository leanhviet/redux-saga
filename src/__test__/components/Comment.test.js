// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

// Component
import Comment from '../../components/Comment'

const mockProps = {
  id: 1,
  name: 'Le Dinh Nam',
  src: '',
  content:
    'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Minus quibusdam omnis iusto laboriosam. Nulla aliquid sit, ipsa blanditiis, minus labore asperiores quia tempore culpa dignissimos necessitatibus quod distinctio facilis magnam!',
  likeNumber: '150',
  dislikeNumber: '300'
}

const wrapper = (props = {}) => {
  const component = shallow(<Comment comment={props} />)

  return component
}

describe('Comment component', () => {
  const wrapperWithoutProps = wrapper()

  // Snapshot
  test('should render correctly Comment component default props', () => {
    expect(shallowToJson(wrapperWithoutProps)).toMatchSnapshot()
  })

  // Testing props
  describe('Handle with props', () => {
    test('should is rendered default props', () => {
      expect(wrapperWithoutProps.find('Profile').props().src).toBeFalsy()
      expect(wrapperWithoutProps.find('Profile').props().name).toMatch(
        'Le Anh Viet'
      )
      expect(wrapperWithoutProps.find('Profile').props().logout).toBeFalsy()
    })

    test('should is rendered with a custom Props', () => {
      const wrapperWithProps = wrapper(mockProps)

      expect(wrapperWithProps.find('Profile').props().src).toMatch(
        mockProps.src
      )
      expect(wrapperWithProps.find('Profile').props().name).toMatch(
        mockProps.name
      )
      expect(wrapperWithProps.find('Profile').props().subName).toMatch(
        mockProps.content
      )
      expect(wrapperWithProps.find('Profile').props().logout).toBeFalsy()
    })
  })
})
