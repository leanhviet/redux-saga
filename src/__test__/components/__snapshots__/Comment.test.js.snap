// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component should render correctly Comment component default props 1`] = `
<div
  className="comment"
>
  <Profile
    name="Le Anh Viet"
    onLogout={[Function]}
    src=""
    subName="9 Subscribers"
  />
  <div
    className="text-right comment__menu"
  >
    <span
      className="comment__menu__icon"
    >
      <i
        className="fa fa-thumbs-up"
      />
    </span>
    <span
      className="comment__menu__icon"
    >
      <i
        className="fa fa-thumbs-down"
      />
    </span>
    <span
      className="comment__menu__icon"
    >
      reply
    </span>
  </div>
</div>
`;
